jenkins/opensearch/Jenkinsfile

```diff
@@ -174,6 +176,21 @@ void build() {
     lib.Messages.new(this).add("${STAGE_NAME}", "${baseUrl}/builds/manifest.yml\n${baseUrl}/dist/manifest.yml")
 }
 
+void test() {
+    def buildManifest = lib.BuildManifest.new(readYaml(file: 'builds/manifest.yml'))
+
+    def baseUrl = buildManifest.getArtifactRootUrl(env.PUBLIC_ARTIFACT_URL, env.JOB_NAME, env.BUILD_NUMBER)
+    def artifactPath = buildManifest.getArtifactRoot(env.JOB_NAME, env.BUILD_NUMBER)
+
+    sh "./test.sh integ-test ${baseUrl} --test-run-id ${env.BUILD_NUMBER}"
+
+    echo "Uploading Test Results to s3://${ARTIFACT_BUCKET_NAME}/${artifactPath}/test-results"
+    withAWS(role: 'opensearch-test', roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
+        s3Upload(file: 'test-results', bucket: "${ARTIFACT_BUCKET_NAME}", path: "${artifactPath}/test-results")
+    }
+
+}
+
 /** Publishes a notification to a slack instance*/
 void publishNotification(icon, msg, extra) {
```

Comment:
Need help: How can we handle Notification of the build and test run together here https://github.com/opensearch-project/opensearch-build/pull/972/files#diff-d2e59eb1985423c8e84c1b54324763f503febaea168a3fab1f44339b754cac0fR149?

Reply:
For example: `OpenSearch 1.3.0 #1231 has built and integ test failed`

Reply:
`lib.Messages.new(this).add("${STAGE_NAME}", "${baseUrl}/builds/manifest.yml\n${baseUrl}/dist/manifest.yml")` is how to append output to the notifications.

If you wanted to check for pass/failure of tests, you might want to add a message that you can read.  There are some limitations with the message passing, so you'll need to append a '-test' or something to the `"${STAGE_NAME}"` and update the message fetching logic

When you are in the success / failure nodes you can attempt to read the messages `def stashed = lib.Messages.new(this).get(['build-x64', 'build-arm64'])` and take additional action.